src/slice.cr

```diff
@@ -119,6 +119,30 @@ struct Slice(T)
     new(size, read_only: read_only) { value }
   end
 
+  # Creates a `Slice` to the given *pointer*, bounded by a null-terminator. This
+  # method does not allocate heap memory.
+  #
+  # The use of a 'reasonable' value for `limit` is strongly recommended. This will
+  # prevent the function from unlimited reading of unintended memory.
+  #
+  # The limit is inclusive. That is to say the limit character will be included, but
+  # no further characters will.
+  #
+  # ```
+  # ptr = LibC.getgrnam("wheel").value.gr_mem
+  # slice = Slice.new(ptr, limit: LibC::NGROUP_MAX, read_only: true)
+  # ```
+  def self.new(pointer : U**, *, limit : Int, read_only = false) forall U
```

Comment:
Can you annotate the return type here? Should be `Slice(U*)`.

Reply:
Fixed